diff --git a/js/src/vm/JSONPrinter.cpp b/js/src/vm/JSONPrinter.cpp
--- a/js/src/vm/JSONPrinter.cpp
+++ b/js/src/vm/JSONPrinter.cpp
@@ -200,7 +200,7 @@ void JSONPrinter::property(const char* name, uint64_t value) {
   out_.printf("%" PRIu64, value);
 }

-#if defined(XP_DARWIN) || defined(__OpenBSD__) || defined(__wasi__)
+#if defined(XP_DARWIN) || defined(__OpenBSD__) || defined(__wasi__) || defined(__wasm32__)
 void JSONPrinter::property(const char* name, size_t value) {
   propertyName(name);
   out_.printf("%zu", value);
diff --git a/js/src/vm/JSONPrinter.h b/js/src/vm/JSONPrinter.h
--- a/js/src/vm/JSONPrinter.h
+++ b/js/src/vm/JSONPrinter.h
@@ -53,7 +53,7 @@ class JSONPrinter {
   void property(const char* name, uint32_t value);
   void property(const char* name, int64_t value);
   void property(const char* name, uint64_t value);
-#if defined(XP_DARWIN) || defined(__OpenBSD__) || defined(__wasi__)
+#if defined(XP_DARWIN) || defined(__OpenBSD__) || defined(__wasi__) || defined(__wasm32__)
   // On OSX and OpenBSD, size_t is long unsigned, uint32_t is unsigned, and
   // uint64_t is long long unsigned. Everywhere else, size_t matches either
   // uint32_t or uint64_t.
